### ISSUE TITLE
podman log messages do not print carriage return when using --tty flag

### ISSUE THREAD

**Issue by marcov** [projectatomic/libpod#961]
<!--
---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Is this a BUG REPORT or FEATURE REQUEST?**:
> /kind bug

**Description**

When using podman to run a container with the --tty flag, the log messages printed with the --log-level option do not include the carriage return.

**Steps to reproduce the issue:**

```podman --log-level info run --tty --rm hello-world```


**Describe the results you received:**
The following output:
```
INFO[0000] [graphdriver] using prior storage driver: overlay 
INFO[0000] CNI network podman (type=bridge) is used from /etc/cni/net.d/87-podman-bridge.conflist 
INFO[0000] Initial CNI setting succeeded                
INFO[0000] Got pod network {Name:goofy_brattain Namespace:goofy_brattain ID:910c99655facae46d20557cea8037f61a3e2907ccf66f173e31612b48eff33aa NetNS:/var/run/netns/cni-eb5d87f5-6c07-1182-e518-4d6a0aafb9e2 PortMappings:[]} 
                                                                                                                                                                                                                            INFO[0000] About to add CNI network cni-loopback (type=loopback) 
                                                INFO[0000] Got pod network {Name:goofy_brattain Namespace:goofy_brattain ID:910c99655facae46d20557cea8037f61a3e2907ccf66f173e31612b48eff33aa NetNS:/var/run/netns/cni-eb5d87f5-6c07-1182-e518-4d6a0aafb9e2 PortMappings:[]} 
                               INFO[0000] About to add CNI network podman (type=bridge) 
                                                                                        WARN[0001] failed to parse language "en_US.utf8": language: tag is not well-formed 

Hello from Docker!
This message shows that your installation appears to be working correctly.
```

**Describe the results you expected:**
The following output:
```
INFO[0000] [graphdriver] using prior storage driver: overlay 
INFO[0000] CNI network podman (type=bridge) is used from /etc/cni/net.d/87-podman-bridge.conflist 
INFO[0000] Initial CNI setting succeeded                
INFO[0000] Got pod network {Name:goofy_brattain Namespace:goofy_brattain ID:910c99655facae46d20557cea8037f61a3e2907ccf66f173e31612b48eff33aa NetNS:/var/run/netns/cni-eb5d87f5-6c07-1182-e518-4d6a0aafb9e2 PortMappings:[]} 
                                                                                                                                                                                                                            INFO[0000] About to add CNI network cni-loopback (type=loopback) 
INFO[0000] Got pod network {Name:goofy_brattain Namespace:goofy_brattain ID:910c99655facae46d20557cea8037f61a3e2907ccf66f173e31612b48eff33aa NetNS:/var/run/netns/cni-eb5d87f5-6c07-1182-e518-4d6a0aafb9e2 PortMappings:[]} 
INFO[0000] About to add CNI network podman (type=bridge) 
WARN[0001] failed to parse language "en_US.utf8": language: tag is not well-formed 

Hello from Docker!
This message shows that your installation appears to be working correctly.
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:       0.6.3
Go Version:    go1.10.3
OS/Arch:       linux/amd64
```

**Output of `podman info`:**

```
host:
  MemFree: 4760469504
  MemTotal: 16700174336
  SwapFree: 16868438016
  SwapTotal: 16869486592
  arch: amd64
  cpus: 4
  hostname: wildfox
  kernel: 4.17.1-1-default
  os: linux
  uptime: 19h 49m 25.02s (Approximately 0.79 days)
insecure registries:
  registries: []
registries:
  registries:
  - docker.io
store:
  ContainerStore:
    number: 4
  GraphDriverName: overlay
  GraphOptions: []
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
  ImageStore:
    number: 5
  RunRoot: /var/run/containers/storage
```

**Comment by mheon** [comment]
This is a side effect of setting raw mode on the terminal. Would definitely be nice to fix.

**Comment by rhatdan** [comment]
Is this a recent change?

Or is this something we need to change in logrus?

**Comment by mheon** [comment]
I don't think anyone has bothered looking at this in a while. Could be a simple change in logrus, or even a config option we're not setting?

**Comment by marcov** [comment]
A way to fix this is to alternate between the default logrus text formatter and a raw text formatter whenever the terminal is in raw mode. 

The raw text formatter can just append an extra ```\r``` to the default text formatter: 
```go
type RawTtyFormatter struct {
}

func (f *RawTtyFormatter) Format(entry *logrus.Entry) ([]byte, error) {
	textFormatter := logrus.TextFormatter{}
	bytes, err := textFormatter.Format(entry)

	if err == nil {
		bytes = append(bytes, '\r')
	}

	return bytes, err
}
````
I tested this locally and it looks good. Do you want me to submit a PR?

**Comment by baude** [comment]
yes please

**Comment by mheon** [comment]
Closed by #967 - thanks for fixing this @marcov
